homeassistant/components/mqtt/__init__.py

```diff
@@ -364,18 +364,27 @@ async def finish_dump(_):
     hass.data[DATA_CONFIG_ENTRY_LOCK] = asyncio.Lock()
     hass.data[CONFIG_ENTRY_IS_SETUP] = set()
 
-    async def async_forward_entry_setup():
+    async def async_forward_entry_setup(config_entry):
         """Forward the config entry setup to the platforms."""
-        async with hass.data[DATA_CONFIG_ENTRY_LOCK]:
-            for component in PLATFORMS:
-                config_entries_key = f"{component}.mqtt"
-                if config_entries_key not in hass.data[CONFIG_ENTRY_IS_SETUP]:
-                    hass.data[CONFIG_ENTRY_IS_SETUP].add(config_entries_key)
-                    await hass.config_entries.async_forward_entry_setup(
-                        entry, component
-                    )
-
-    hass.async_create_task(async_forward_entry_setup())
+        # Local import to avoid circular dependencies
+        # pylint: disable-next=import-outside-toplevel
+        # pylint: disable-next=import-outside-toplevel
+        from . import device_automation, tag
```

Comment:
```suggestion
        # Local import to avoid circular dependencies
        # pylint: disable-next=import-outside-toplevel
        from . import device_automation, tag
```